File: FactorialRecursion.java
public class FactorialRecursion {

// Recursive function to calculate factorial
    public static long factorial(int n) {
        if (n == 0 || n == 1) {
          return 1; // Base case
       }
        return n * factorial(n - 1); // Recursive call
    }

    public static void main(String[] args) {
       int number = 5;
       long result = factorial(number);

       System.out.println("Factorial of " + number + " is: " + result);
    }
}

File: factorial_recursion.py
def factorial(n):
    """Calculates the factorial of a number using recursion."""
    if n == 0 or n == 1:
        return 1  # Base case: 0! and 1! are 1
    
    # Recursive call: n! = n * (n-1)!
    return n * factorial(n - 1)

if __name__ == "__main__":
    number = 5
    result = factorial(number)
    print(f"Factorial of {number} is: {result}")